tests/sentry/issues/endpoints/test_organization_group_index.py

```diff
@@ -4418,27 +4403,81 @@ def test_delete_performance_issue_by_id(self, mock_eventstream: MagicMock) -> No
 
         self.assert_deleted_groups([group1, group2])
 
-    def test_bulk_delete(self) -> None:
-        groups = self.create_n_groups(20)
-
-        self.login_as(user=self.user)
-        response = self.get_success_response(qs_params={"query": ""})
-        assert response.status_code == 204
-        self.assert_pending_deletion_groups(groups)
+    def test_bulk_delete_for_many_projects(self) -> None:
+        NEW_CHUNK_SIZE = 2
 
-        # This is needed to put the groups in the unresolved state before also triggering the task
-        Group.objects.filter(id__in=[group.id for group in groups]).update(
-            status=GroupStatus.UNRESOLVED
-        )
-
-        with self.tasks():
-            response = self.get_success_response(qs_params={"query": ""})
-            assert response.status_code == 204
-
-        self.assert_deleted_groups(groups)
+        with self.feature("organizations:global-views"):
+            project_2 = self.create_project(slug="baz", organization=self.organization)
+            groups_1 = self.create_n_groups_with_hashes(2, project=self.project)
+            groups_2 = self.create_n_groups_with_hashes(5, project=project_2)
+
+            with (
+                self.tasks(),
+                patch("sentry.deletions.tasks.groups.GROUP_CHUNK_SIZE", NEW_CHUNK_SIZE),
+                patch("sentry.deletions.tasks.groups.logger") as mock_logger,
+                patch(
+                    "sentry.api.helpers.group_index.delete.uuid4",
+                    side_effect=[self.get_mock_uuid("foo"), self.get_mock_uuid("bar")],
+                ),
+            ):
+                self.login_as(user=self.user)
+                response = self.get_success_response(qs_params={"query": ""})
+                assert response.status_code == 204
+                batch_1 = [g.id for g in groups_2[0:2]]
+                batch_2 = [g.id for g in groups_2[2:4]]
+                batch_3 = [g.id for g in groups_2[4:]]
+                assert batch_1 + batch_2 + batch_3 == [g.id for g in groups_2]
+
+                # Each project is scheduled without a transaction_id, so we expect the same transaction_id for both calls
+                assert mock_logger.info.call_args_list == [
+                    call(
+                        "delete_groups.started",
+                        extra={
+                            "object_ids_count": 5,
```

Comment:
The first-time project 2 gets scheduled, it has the ids of all the groups.

Each subsequent task will have `NEW_CHUNK_SIZE` less ids.